ipykernel/kernelbase.py

```diff
@@ -1132,7 +1168,12 @@ async def list_subshell_request(self, socket, ident, parent) -> None:
 
     async def apply_request(self, socket, ident, parent):  # pragma: no cover
         """Handle an apply request."""
-        self.log.warning("apply_request is deprecated in kernel_base, moving to ipyparallel.")
+        warnings.warn(
```

Comment:
Since the async changes totally break all IPython Parallel versions that didn't have these methods, it probably makes sense to remove the IPython Parallel methods in ipykernel 7, as keeping them while breaking them at the same time doesn't make a lot of sense.

Methods/attributes to remove:

- `aborted`
- `_send_abort_reply`
- `apply_request`
- `abort_request`
- `clear_request`
- `do_clear`

I can work on that, if you want, since I'm working on the ipyparallel updates now anyway

Reply:
done in #1282